pandera/io.py

```diff
@@ -25,15 +26,23 @@
 NOT_JSON_SERIALIZABLE = {PandasDtype.DateTime, PandasDtype.Timedelta}
 
 
-def _serialize_check_stats(check_stats, pandas_dtype):
+def _serialize_check_stats(check_stats, pandas_dtype=None):
     """Serialize check statistics into json/yaml-compatible format."""
+    # pylint: disable=unused-argument
 
     def handle_stat_dtype(stat):
+        # fixme: change interface to not require a dtype spec
```

Comment:
I assume this is fixed with `pandas_dtype=None` above?

Reply:
I was worried that my fix is a hack and that there's a better way to do this (I don't like nonlocal). In the component-wise case it's an optimization that prevents a dtype lookup because you assume that each component has the same types for the statistics. Is this a valid assumption? Maybe we can just drop the dtype argument.

Reply:
I've committed a change that circumvents `nonlocal` (not a big fan either). I also noticed that the dtype inference could fail when one stat was not a recognized dtype. More details in the general discussion.

An example of failure was:

```python
import pandera as pa

schema = pa.DataFrameSchema(
    columns={
        "notype_column": pa.Column(
            str,
            checks=pa.Check.isin(["foo", "bar", "x", "xy"]), # list is not a PandasDtype
        )
    }
)
schema.to_yaml()
```